pkg/stanza/fileconsumer/internal/fingerprint/fingerprint.go

```diff
@@ -18,50 +19,64 @@ const MinSize = 16 // bytes
 // Fingerprint is used to identify a file
 // A file's fingerprint is the first N bytes of the file
 type Fingerprint struct {
-	FirstBytes []byte
+	firstBytes  []byte
+	HashBytes   []byte
+	BytesLength int
 }
 
 // New creates a new fingerprint from an open file
 func New(file *os.File, size int) (*Fingerprint, error) {
 	buf := make([]byte, size)
-
 	n, err := file.ReadAt(buf, 0)
 	if err != nil && !errors.Is(err, io.EOF) {
 		return nil, fmt.Errorf("reading fingerprint bytes: %w", err)
 	}
+	fBytes := buf[:n]
+
+	h := fnv.New128a()
+	h.Write(fBytes)
+	hash := h.Sum(nil)
 
 	fp := &Fingerprint{
-		FirstBytes: buf[:n],
+		firstBytes:  fBytes,
+		HashBytes:   hash,
+		BytesLength: len(fBytes),
 	}
 
 	return fp, nil
 }
 
 // Copy creates a new copy of the fingerprint
 func (f Fingerprint) Copy() *Fingerprint {
-	buf := make([]byte, len(f.FirstBytes), cap(f.FirstBytes))
-	n := copy(buf, f.FirstBytes)
+	buf := make([]byte, len(f.firstBytes), cap(f.firstBytes))
+	n := copy(buf, f.firstBytes)
 	return &Fingerprint{
-		FirstBytes: buf[:n],
+		firstBytes:  buf[:n],
+		HashBytes:   f.HashBytes,
+		BytesLength: f.BytesLength,
+	}
+}
+
+// UpdateFingerPrint will update fingerprint with new bytes
+func (f *Fingerprint) UpdateFingerPrint(offset int64, appendBytes []byte) {
+	if f.firstBytes == nil {
+		f.firstBytes = appendBytes
+	} else {
+		f.firstBytes = append(f.firstBytes[:offset], appendBytes...)
 	}
+	h := fnv.New128a()
+	h.Write(f.firstBytes)
+	hash := h.Sum(nil)
+	f.HashBytes = hash
+	f.BytesLength = len(f.firstBytes)
 }
 
 // Equal returns true if the fingerprints have the same FirstBytes,
 // false otherwise. This does not compare other aspects of the fingerprints
 // because the primary purpose of a fingerprint is to convey a unique
 // identity, and only the FirstBytes field contributes to this goal.
```

Comment:
```suggestion
 FirstBytes field contributes to this goal.
```
this comment does not seem accurate anymore